results/cmd/watcher/main_test.go

```diff
@@ -0,0 +1,121 @@
+	"knative.dev/pkg/logging"
+)
+
+var (
+	srv *server.Server
+)
+
+func getRunName(tr *v1beta1.TaskRun) string {
+	return strings.Join([]string{tr.Namespace, tr.Name}, "/")
+}
+
+// getFakeClients create a fake client to send test data to reconciler
+func getFakeClients(t *testing.T, tr []*v1beta1.TaskRun) (test.Assets, context.CancelFunc) {
+	t.Helper()
+	ctx, _ := ttesting.SetupFakeContext(t)
+	ctx, cancel := context.WithCancel(ctx)
+	d := test.Data{
+		TaskRuns: tr,
+	}
+	clients, _ := test.SeedTestData(t, ctx, d)
+	cmw := configmap.NewInformedWatcher(clients.Kube, "")
+	return test.Assets{
+		Controller: newController(ctx, cmw, t),
+		Clients:    clients,
+	}, cancel
+}
+
+// newController creates a Controller with provided context and configmap
+func newController(ctx context.Context, cmw configmap.Watcher, t *testing.T) *controller.Impl {
+	logger := logging.FromContext(ctx)
+	taskRunInformer := taskruninformer.Get(ctx)
+	pipelineclientset := pipelineclient.Get(ctx)
+	c := &reconciler{
+		logger:            logger,
+		taskRunLister:     taskRunInformer.Lister(),
+		fakeserver:        srv,
+		pipelineclientset: pipelineclientset,
+	}
+	impl := controller.NewImpl(c, c.logger, pipeline.PipelineRunControllerName)
+	taskRunInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    impl.Enqueue,
+		UpdateFunc: controller.PassNew(impl.Enqueue),
+	})
+	return impl
+}
+
+// getNewTaskRun sends TaskRun data to reconciler and then retrieves completed TaskRun
+func getNewTaskRun(taskRun v1beta1.TaskRun, d []*v1beta1.TaskRun, t *testing.T) (*v1beta1.TaskRun, error) {
+	testAssets, cancel := getFakeClients(t, d)
+	defer cancel()
+	c := testAssets.Controller
+	clients := testAssets.Clients
+	if err := c.Reconciler.Reconcile(context.Background(), getRunName(&taskRun)); err != nil {
+		return nil, err
+	}
+	tr, err := clients.Pipeline.TektonV1beta1().TaskRuns(taskRun.Namespace).Get(taskRun.Name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return tr, nil
+}
+
+// TestReconcile tests if TaskRun in the client can be properly updated when sent to reconciler
+func TestReconcile(t *testing.T) {
+	server, err := server.SetupTestDB(t)
+	if err != nil {
+		t.Fatalf("Failed to create fake server: %v", err)
+	}
+	srv = server
+	taskRun := v1beta1.TaskRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        "Tekton",
+			Namespace:   "default",
+			Annotations: map[string]string{"demo": "demo"},
+		},
+	}
+	d := []*v1beta1.TaskRun{
+		&taskRun,
+	}
+	// Create a new TaskRun in the fake client, completed TaskRun should be updated with a results id field in annotations.
+	tr, err := getNewTaskRun(taskRun, d, t)
+	if err != nil {
+		t.Fatalf("Failed to get completed TaskRun %s: %v", taskRun.Name, err)
+	}
+	_, ok := tr.Annotations["results.tekton.dev/id"]
+	if !ok {
+		t.Fatalf("Expected completed TaskRun %s should be updated with a results_id field in annotations", taskRun.Name)
+	}
+	d = []*v1beta1.TaskRun{
+		tr,
+	}
```

Comment:
Looks like we're only ever calling this with 1 taskrun. Let's move the list wrapping into the func to make it cleaner.

Reply:
done